doc/scenario.md

```diff
@@ -0,0 +1,439 @@
+# Role Scenario
+
+The role uses a scenario to determine what to do. Scenario consists of steps that can be configured.
+
+Using a scenario of steps, you can:
+
+- run the specified steps in any order, but API must be followed (with `cartridge_scenario`);
+- replace the steps of the role with your own or add new steps
+  (with `cartridge_custom_steps_dir` or `cartridge_custom_steps`).
+
+## Steps
+
+In `cartridge_scenario` you should specify names of steps. The default scenario includes the following steps:
+
+- [deliver_package](#deliver_package)
+- [update_package](#update_package)
+- [update_instance](#update_instance)
+- [configure_instance](#configure_instance)
+- [restart_instance](#restart_instance)
+- [wait_instance_started](#wait_instance_started)
+- [connect_to_membership](#connect_to_membership)
+- [edit_topology](#edit_topology)
+- [cleanup_expelled](#cleanup_expelled)
+- [configure_auth](#configure_auth)
+- [configure_app_config](#configure_app_config)
+- [bootstrap_vshard](#bootstrap_vshard)
+- [configure_failover](#configure_failover)
+
+There are additional steps that are not included in the default scenario, but can be used in a custom one:
+
+- [set_control_instance](#set_control_instance)
+
+To replace the steps of the role with your own or add new steps, you should use `cartridge_custom_steps_dir`
+or `cartridge_custom_steps` options.
+
+## Examples
+
+### Adding custom step to scenario
+
+You can create your own step and include it to scenario. Let's create `./custom_steps/special_magic.yml`. After that,
+you can add `special_magic` step to scenario. Just pass path to a directory where your custom steps are placed:
+
+```yaml
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
+        cartridge_scenario:
+          - restart_instance
+          - special_magic
+```
+
+### Importing steps from different directories
+
+If you want to import steps by full paths, `cartridge_custom_steps` parameter can be useful. It allows specifying
+mapping between step name and path to file to import:
+
+```yaml
+cartridge_custom_steps:
+  - name: 'common_special_task'
+    file: './common/special_task.yml'
+```
+
+### Override task from default scenario
+
+Sometimes we need to use default scenario, but change some steps realizations.
+
+For example, sometimes we need to deliver package to remote machine by our own way. By default, package specified
+by `cartridge_package_path` is simple copied to remote host, it's happened on `deliver_package` step. Sometimes we need
+to download or copy package from some custom place. Let's do it!
```

Comment:
`For example, sometimes... . Sometimes ... <and the same as before>` - please fix logic.